### ISSUE TITLE
Distinguish between versions of Visual Studio to install

### ISSUE THREAD

**Issue by mrlacey** [microsoft/devhome#771]
### Suggested new feature or improvement

Don't recommend installing a SKU of VS2022 when another is already installed

### Scenario

on the install applications section of the "end-to-end" setup it suggested "Visual Studio 2022 community Edition"
But, I already have the Enterprise and Preview versions of VS2022 installed.

### Additional details

At first I thought this was because there was a new update to the version of VS2022 Enterprise that I hadn't yet installed but that's not the case.

**Comment by florelis** [comment]
See https://github.com/microsoft/devhome/issues/788#issuecomment-1561687716

**Comment by florelis** [comment]
@denelon , maybe this could be done with something like virtual packages on the package management side instead of configuration? We could have packages that we count as installed if one of a list of others are installed. So, we could have a VisualStudio package that counts as installed if we have any of Community/Pro/Enterprise. It would also help with stuff like just wanting to have a JRE/JDK, or any Python 3. It feels like we could do this in a pretty clean way

**Comment by denelon** [comment]
Yes, thanks @florelis for bringing this into the discussion as well.

That might help reduce the potential proliferation of changes across the myriad of manifests that have dependencies on languages, runtimes, etc. It's similar to the "provider"/"consumer" model in some other package managers.
